pkl-cli/src/main/kotlin/org/pkl/cli/commands/FormatterCommand.kt

```diff
@@ -26,75 +25,75 @@ import com.github.ajalt.clikt.parameters.options.option
 
-class FormatterCommand : NoOpCliktCommand(name = "format") {
-  override fun help(context: Context) = "Run commands related to formatting"
+class FormatterCommand : CliktCommand(name = "format") {
+  override fun help(context: Context) =
+    """
+    Format or check formatting of Pkl files.
+    
+    Examples:
+    
+    ```
+    # Overwrite all Pkl files inside `my/folder/`, recursively.
+    $ pkl format -w my/folder/
+    
+    # Check formatting of all files, printing filenames with formatting violations to stdout.
+    # Exit with exit code `11` if formatting violations were found.
+    $ pkl format --names my/folder/
+    
+    # Format Pkl code from stdin.
+    $ echo "foo = 1" | pkl format -
+    ```
+  """
+      .trimIndent()
 
   override fun helpEpilog(context: Context) = "For more information, visit $helpLink"
 
-  init {
-    subcommands(FormatterCheckCommand(), FormatterApplyCommand())
-  }
-}
-
-class FormatterCheckCommand : BaseCommand(name = "check", helpLink = helpLink) {
-  override val helpString: String =
-    "Check if the given files are properly formatted, printing the file name to stdout in case they are not. Returns non-zero in case of failure."
-
   val paths: List<Path> by
-    argument(name = "paths", help = "Files or directory to check.")
-      .path(mustExist = true, canBeDir = true)
+    argument(name = "paths", help = "Files or directory to check. Use `-` to read from stdin.")
+      .path(mustExist = false, canBeDir = true)
       .multiple()
 
   val grammarVersion: GrammarVersion by
-    option(names = arrayOf("--grammar-version"), help = grammarVersionHelp)
+    option(
+        names = arrayOf("--grammar-version"),
+        help =
+          """
+          The grammar compatibility version to use.$NEWLINE
+          ${GrammarVersion.entries.joinToString("$NEWLINE", prefix = "  ") {
+            val default = if (it == GrammarVersion.latest()) " `(default)`" else ""
+            "`${it.version}`: ${it.versionSpan}$default"
+          }}
+          """
+            .trimIndent(),
+      )
       .enum<GrammarVersion> { "${it.version}" }
       .default(GrammarVersion.latest())
 
-  override fun run() {
-    CliFormatterCheck(baseOptions.baseOptions(emptyList()), paths, grammarVersion).run()
-  }
-
-  companion object {
-    internal val grammarVersionHelp =
-      """
-      The grammar compatibility version to use.$NEWLINE
-      ${GrammarVersion.entries.joinToString("$NEWLINE", prefix = "  ") {
-        val default = if (it == GrammarVersion.latest()) " `(default)`" else ""
-        "`${it.version}`: ${it.versionSpan}$default"
-      }}
-      """
-        .trimIndent()
-  }
-}
-
-class FormatterApplyCommand : BaseCommand(name = "apply", helpLink = helpLink) {
-  override val helpString: String =
-    "Overwrite all the files in place with the formatted version. Returns non-zero in case of failure."
+  val overwrite: Boolean by
+    option(
+        names = arrayOf("-w", "--write"),
```

Comment:
`-w` conflicts with the shorthand for `--working-dir`!

Reply:
Ah, I missed that this inherits from `CliktCommand` now and not `BaseCommand` so the regular evaluation options don't get pulled in, unlike the previous iteration.